mockwebserver/src/test/java/mockwebserver3/MockResponseSniTest.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2022 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package mockwebserver3
+
+import okhttp3.Dns
+import okhttp3.Headers.Companion.headersOf
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClientTestRule
+import okhttp3.Request
+import okhttp3.testing.PlatformRule
+import okhttp3.tls.HandshakeCertificates
+import okhttp3.tls.HeldCertificate
+import okhttp3.tls.internal.TlsUtil.localhost
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+
+class MockResponseSniTest {
+  @RegisterExtension
+  val clientTestRule = OkHttpClientTestRule()
+
+  @RegisterExtension
+  val platform = PlatformRule()
+
+  private lateinit var server: MockWebServer
+
+  @BeforeEach
+  fun setUp(server: MockWebServer) {
+    this.server = server
+  }
+
+  @Test
+  fun clientSendsServerNameAndServerReceivesIt() {
+    // java.net.ConnectException: Connection refused
```

Comment:
I should look into this, but this test is better than no test, as it confirms our current behaviour.